scripts/basic-workflow-tests.sh

```diff
@@ -74,6 +73,35 @@ function init_e2e_tests() {
 }
 
 
+function wait_for_server() {
```

Comment:
Also, it seems this move of the `wait_for_server` function is not needed. I see the commit message suggests that bash LSP fails to detect this as a valid shell code? It is though, so probably a bug in the LSP used? I don't think shell requires a particular order of functions declared as long as they are not executed before being declared.

Reply:
What's the problem with moving it in a way that makes it easier for people to develop and work with the file?